.jenkinsfile

```diff
@@ -1,113 +1,243 @@
+
+        // Perform checkout in specific location.
+        // The relative folder structure is important for the ctest script.
+        dir('checkout')
+        {
+            checkout scm
+        }
     }
-    stages {
-        stage ('Configure') { // TODO(rrahn): Check if really needed.
-            steps {
-                script {
-                    echo "The pushed branch: ${BRANCH_NAME}"
-                }
+
+    // Prepare environment and run ctest step.
+    withEnv(["BUILDNAME=$build_name",
+             "PLATFORM=$platform",
+             "MODEL=$model",
+             "TEST_MODEL=unit",
+             "WORKSPACE=$workspace",
+             "SITE_NAME=jenkins",
+             "DISABLE_CEREAL=$disable_cereal",
+             "CTEST_BUILD_TYPE=$build_type",
+             "CXX=$cxx"])
+    {
+        stage('run')
+        {
+            dir(workspace)
+            {
+                sh "ctest --no-compress-output -VV -S $workspace/seqan3-infra/ctest/seqan3_jenkins.cmake"
             }
         }
-        stage ('Test') { // Perform CI tests.
-            parallel {
-                stage ('unit') {
-                    agent none
-                    steps {
-                        build job: 'pipeline_unit_tests_unix', parameters: [string(name: 'COMMIT_SHA', value: "${GIT_COMMIT}")], quietPeriod: 0
-                    }
-                }
-                stage ('doc') {
-                    agent none
-                    steps {
-                        build job: 'pipeline_doc_tests', parameters: [string(name: 'COMMIT_SHA', value: "${GIT_COMMIT}")], quietPeriod: 0
-                    }
-                }
-                stage ('header') {
-                    agent none
-                    steps {
-                        build job: 'pipeline_header_tests', parameters: [string(name: 'COMMIT_SHA', value: "${GIT_COMMIT}")], quietPeriod: 0
-                    }
-                }
-                stage ('memory') {
-                    steps {
-                        echo "TODO: Perform memory tests."
-                    }
-                }
-                stage ('coverage') {
-                    steps {
-                        echo "TODO: Perform coverage tests."
-                    }
-                }
-                stage ('performance') {
-                    agent none
-                    steps {
-                        build job: 'pipeline_performance_tests', parameters: [string(name: 'COMMIT_SHA', value: "${GIT_COMMIT}")], quietPeriod: 0
+    }
+}
+
+// ============================================================================
+// Define unit test matrix
+// ============================================================================
+
+/* unit builds
+ * compiler: g++-7, g++-8 TODO: g++-9, clang++-concepts
+ * agent: ubuntu TODO macos, freebsd
+ * cereal: true false
+ * build_type: Release Debug
+ */
+
+// Configure linux unit tests
+def axis_agent = ["ununtu"]
```

Comment:
If this stuff is specific to Linux, should it be at the top-level of the file? Souldn't there be some kind of scope? [the language this file in is unknown to me]

Reply:
it is groovy, a java compatible oop language. But as a sandboxed version, so not all expressions can be used (I figured out the hard way)
The file itself is really plain. Groups in this sense are made by stages which define a set of steps to be executed. Since we want the jobs to be executed in parallel there is this matrix stage which will trigger all predefined tasks in a parallel environment

Reply:
> And is this 1604 or 1804 now?

Both now

Reply:
But in the sense of which one is available. We have only 4 executors for 1804 and 4 for 1604, so we can get more parallel builds.